### PR TITLE
More Windows installer changes

### LINKED ISSUE #565: Make installation of VC runtime optional on Windows installer
The powershell install script for Windows supports now to supply a previously downloaded zip file ("archive") of the IoT Edge daemon. But it will still try to download and install every the VC runtime.
It should be either

- made optional to download and install VC at all
- or at least, give the user the chance to manually supply the path to the downloaded installer.exe similar to the IoT Edge daemon.

Especially when re-installing Edge, the VC is obviously already there and downloading it again just takes unnecessary time. Or - even worse - in cases where the device does not have internet connectivity, even behind a proxy, this might fail.

Still seems the case in the latest PR regarding the installer: https://github.com/Azure/iotedge/pull/550/files#diff-8ae7a24c87a2048c653a1cfb9f7e1d8bR502

### PR DESCRIPTION
**Pull request by arsing**
- Add new `-InvokeWebRequestParameters` parameter. This can be used to specify
  extra options for all `Invoke-WebRequest` calls.

- Add new `-OfflineInstallationPath` parameter for offline installs. This is now set to a directory
  that contains one or more files. For each component that the installer
  downloads (iotedged, Moby Engine, Moby CLI, VC Runtime), it first checks if
  the corresponding file is already present in this directory. If the file
  is there, it uses that instead of downloading from the internet.

- Change Moby CLI from opt-in to opt-out. `-WithMobyCli` is renamed to
  `-SkipMobyCli`

- Change all references to "Moby Runtime" to "Moby Engine", except for the
  key in `config.yaml`

Fixes #565

### CODE REVIEW

**Comment by arsing** [comment]
QuickStart needs to be changed as well since it uses `-ArchivePath`

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 1]
```diff
@@ -68,13 +70,25 @@ function Install-SecurityDaemon {
...
         [SecureString] $Password,
 
-        # Also install the Moby CLI (docker.exe) to $MobyInstallDirectory. Only takes affect for `-ContainerOs Windows`
-        [Switch] $WithMobyCli
+        # Splatted into every Invoke-WebRequest invocation. Can be used to set extra options.
+        #
+        # If -Proxy is also specified, it overrides the `-Proxy` key set in this hashtable, if any.
+        #
+        # Example:
+        #     Install-SecurityDaemon -InvokeWebRequestParameters @{ '-Proxy' = 'http://localhost:8888'; '-ProxyCredential' = (Get-Credential).ToNetworkCredential() }
```
Comment (arsing) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: `s/ToNetworkCredential/GetNetworkCredential/`

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 2]
```diff
@@ -572,30 +590,37 @@ function Reset-SystemPath {
 
 function Get-VcRuntime {
     if (Test-IotCore) {
-        Write-HostGreen 'Skipped vcruntime download on IoT Core.'
+        Write-HostGreen 'Skipping VC Runtime installation on IoT Core.'
```
Comment (myagley) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: I believe there is an issue (#565) that is asking if we can make this optional?

Comment (arsing) [reply scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: https://github.com/Azure/iotedge/pull/592/commits/54da4c803097001edb0a7ec1d022c2a5fac9f0b0

I'll add the `Fixes ` to the commit when I merge it.

File: scripts/windows/setup/IotEdgeSecurityDaemon.ps1 [hunk 3]
```diff
@@ -624,6 +665,7 @@ function Install-Services {
 
 function Uninstall-Services {
     if (Get-Service $EdgeServiceName -ErrorAction SilentlyContinue) {
+        Set-Service -StartupType Disabled $EdgeServiceName -ErrorAction SilentlyContinue
         Stop-Service -NoWait -ErrorAction SilentlyContinue -ErrorVariable cmdErr $EdgeServiceName
```
Comment (damonbarry) [inline scripts/windows/setup/IotEdgeSecurityDaemon.ps1]: I think we need to review how we're uninstalling the service (E.g., Do we still need to use `-NoWait`? Can we get rid of the sleep?). But let's do that separately...

**Pull request merged**